core/src/main/scala/com/salesforce/op/stages/impl/tuning/DataCutter.scala

```diff
@@ -110,26 +111,28 @@ class DataCutter(uid: String = UID[DataCutter]) extends Splitter(uid = uid) with
   private[op] def estimate(labelCounts: DataFrame): (Set[Double], Set[Double]) = {
     val minLabelFract = getMinLabelFraction
     val maxLabels = getMaxLabelCategories
+    val maxDroppedLabelsRetained = getMaxDroppedLabelsRetained
 
     val colCount = labelCounts.columns(1)
     val totalValues = labelCounts.agg(sum(colCount)).first().getLong(0).toDouble
-    val labelsKeep = labelCounts
+    val labelsOrderedByCount = labelCounts
       .filter(r => (r.getLong(1) / totalValues) >= minLabelFract)
       .sort(col(colCount).desc)
-      .take(maxLabels)
-      .map(_.getDouble(0))
-
-    val labelSet = labelsKeep.toSet
-    val labelsDropped = labelCounts.filter(r => !labelSet.contains(r.getDouble(0))).collect().map(_.getDouble(0)).toSet
+      .limit(maxLabels + maxDroppedLabelsRetained)
+      .collect()
 
+    val topLabels = labelsOrderedByCount.map(_.getDouble(0))
+    val labelSet = topLabels.take(maxLabels).toSet
+    val topLabelsDropped = topLabels.slice(maxLabels, maxLabels + maxDroppedLabelsRetained).toSet
     if (labelSet.nonEmpty) {
-      log.info(s"DataCutter is keeping labels: $labelSet and dropping labels: $labelsDropped")
+      log.info(s"DataCutter is keeping labels: $labelSet and dropping lower cardinality labels " +
```

Comment:
Since we're not outputting the entire dropped label set, we should probably include how many total labels were dropped for help debugging in the future. Is there a cheap way to do this other than `labelCounts.count()`?